### DISCUSSION TITLE
a year of active life with EditableColumns

### DISCUSSION THREAD

**Discussion by Sladous** [Laravel-Backpack/community-forum#1162] [Q&A (Help)]
Hi,
I've been using Backpack\EditableColumns with PHP: 8.1.10, Laravel: 10.48.22 for over a year.
EditableColumns is a great idea to improve the user interface. I have several problems with it:

1) Font formatting wrappers have already started working. Unfortunately, it cannot control the column width. Now I'm in a state where the single digit column is as wide as the long text.
It would be very nice if this worked:
'wrapper' => ['class' => 'form-group col-md-6'],

2) I have a localization problem. Instead of the decimal point "." we use ",". EditableColumns ignores the settings in the backpack.

3) Error persists with ->withFakes() in:
vendor\backpack\editable-columns\src\Http\Controllers\Operations\MinorUpdateOperation.php
It falls on:
Call member function withFakes() on bool

I solve it with the following modification:
 protected function getMinorUpdateRow()
 {
 CRUD::setCurrentOperation('list');
 $this->setupConfigurationForCurrentOperation();
 // return CRUD::getEntriesAsJsonForDatatables([$this->minorUpdateEntry->withFakes()], 1, 1)['data'][0] ?? null;
 return null;
 }

Thank you for your time :-)

**Comment by karandatwani92** [comment]
Hey @Sladous 

I checked the fake in our [demo](https://demo.backpackforlaravel.com/admin/editable-monster?fake=true). It's working without any issues. I wonder if you are calling the wrong method. Here is what the [demo repo](https://github.com/Laravel-Backpack/demo/blob/e522ed4c73f612c720068a6eb922eefcac598bcb/app/Http/Controllers/Admin/EditableMonsterCrudController.php#L56) has:
```php
// demo fake editable_text column
CRUD::column('fake-text')
    ->label('Fake Text')
    ->type('editable_text')
    ->fake(true);

// demo fake editable_switch column
CRUD::column('fake-switch')
    ->label('Fake Switch')
    ->type('editable_switch')
    ->fake(true);

// demo fake editable_checkbox column
CRUD::column('fake-checkbox')
    ->label('Fake Checkbox')
    ->type('editable_checkbox')
    ->fake(true);

// demo fake editable_select column
CRUD::column('fake-select')
    ->label('Fake Select')
    ->type('editable_select')
    ->options(['one' => 'One', 'two' => 'Two', 'three' => 'Three'])
    ->fake(true);
```
Further checking other issues.

**Reply by Sladous** [reply]
That's just weird. I don't use fake.
My code:
    protected function setupListOperation()
    {
        $this->crud->addButtonFromView('top', 'IssueItemTop', 'IssueItemTop', 'ending');
        $this->crud->addButtonFromView('line', 'IssueItemLine', 'IssueItemLine', 'beginning');
        $this->crud->addButtonFromView('bottom', 'IssueItemBottom', 'IssueItemBottom', 'ending');
        
        $this->crud->orderBy('row_order','ASC');
        $this->crud->removeButton('create');
        $this->crud->removeButton('update');
        $this->crud->removeButton('delete');

        // CRUD::Column('Stocked')->label('S');
        CRUD::addColumn([
            'name'    => 'stock_name',
            'label'   => 'Zásoba',
            'type'    => 'editable_text',
            'select_on_click'  => true, // select the entire text on click? default: false
            'min_width'        => '350px', // how wide should the column be?
            'wrapper'   => [
                'element' => 'span',
                    'class' => function ($crud, $column, $entry, $related_key) {
                        if ($entry->stock AND $entry->stock->serial_number_YN) {
                            if (is_null($entry->serial_number) OR $entry->serial_number=='VYPLNIT')
                                return 'badge badge-error';
                            // else return 'badge badge-success';
                        }    
                    },
            ],  
			'save_on_focusout' => true,
            'searchLogic' => function ($query, $column, $searchTerm) {
                // chytré vyhledávání: link "switch 16" port

                // najdi text mezi uvozovkami
                $meziUvozovkami = null;
                $pattern = '/"([^"]+)"/'; // Regulární výraz pro nalezení textu mezi uvozovkami
                if (preg_match($pattern, $searchTerm, $matches)) {
                    $meziUvozovkami = $matches[1];
                } 
                // co zbude po vyjmutí textu z uvozovek
                $textBezUvozovek = preg_replace($pattern, '', $searchTerm);
                // jednotlivá slova
                $hledaneRetezce = explode(' ', $textBezUvozovek);
                $hledaneRetezce = array_filter($hledaneRetezce);

                if ($meziUvozovkami) 
                    $hledaneRetezce[] = $meziUvozovkami;

                $query->where(function ($query) use ($hledaneRetezce) {
                    foreach ($hledaneRetezce as $hR) {
                        $query->where(function ($query) use ($hR) {
                            $query->whereHas('stock', function ($q) use ($hR) {
                                $q->where('full_name', 'like', '%' . $hR . '%');
                            })
                            ->orWhere('stock_name','like', '%' . $hR . '%')
                            ->orWhere('stock_text','like', '%' . $hR . '%')
                            ->orWhere('serial_number','like', '%' . $hR . '%');
                            
                        });
                    }
                });
            }
        ]);

        CRUD::addColumn([
            'name'             => 'units',
            'label'            => 'Množství',
            'type'             => 'editable_text',
            'select_on_click'  => true, 
			'save_on_focusout' => true,
        ]);

        CRUD::addColumn([
            'name'             => 'price',
            'label'            => 'Cena',
            // 'type'             => 'editable_number',
            'type'             => 'editable_text',
            'select_on_click'  => true, 
			'save_on_focusout' => true,
        ]);

        CRUD::addColumn([
            'name'             => 'serial_number',
            'label'            => 'Ser. čsl.',
            'type'             => 'editable_text',
            'select_on_click'  => true, 
			'save_on_focusout' => true,
            // 'wrapper' => ['class' => 'form-group col-md-2'],
        ]);

        CRUD::addColumn([
            'name'             => 'stock_text',
            'label'            => 'Text',
            'type'             => 'editable_text',
            'select_on_click'  => true, 
			'save_on_focusout' => true,
        ]);

        CRUD::addColumn([
            'name'             => 'percentage_discount',
            'label'            => 'Sleva',
            'type'             => 'editable_text',
            'select_on_click'  => true, 
			'save_on_focusout' => true,
        ]);

        CRUD::addColumn([
            'name'             => 'row_order',
            'label'            => 'Poř',
            'type'             => 'editable_text',
            'select_on_click'  => true, 
			'save_on_focusout' => true,
        ]);

        CRUD::addColumn([
            'name'             => 'final_price',
            'label'            => 'Konečná',
            'type'             => 'editable_text',
            'select_on_click'  => true, 
			'save_on_focusout' => true,
        ]);

    }

**Comment by karandatwani92** [comment]
Hey @Sladous 

Can you give more info on your 2nd point to understand it better?

**Reply by Sladous** [reply]
In the USA, the format of the amount is: $1000.56 but here in the Czech Republic: 1000,56 
Did I write it more clearly?

**Comment by tomsb** [comment]
About the @Sladous' 3rd point `Error: Call to a member function withFakes() on true` - I believe there is a bug in the [Add-On readme documentation](https://backpackforlaravel.com/products/editable-columns/readme.md).

Correct way to solve it (because `->save()` returns a boolean, but `getMinorUpdateRow` needs the entry itself):

```diff
public function saveMinorUpdateEntry()
{
    // Save the minor update
    $entry = $this->crud->getModel()->find(request('id'));
    $entry->{request('attribute')} = request('value');
    
    // Perform specific logic
    $entry->status = 'draft';

-    return $entry->save();
+    return $entry;
+    // Alternatively, you can return a fresh entry if needed
+    // return $entry->refresh();
}
```

**Reply by Sladous** [reply]
I checked, but the suggested change doesn't work.
